pkg/kubectl/resource/builder_test.go

```diff
@@ -899,11 +899,11 @@ func TestMultipleObject(t *testing.T) {
 		t.Fatalf("unexpected error: %v", err)
 	}
 
-	expected := &api.List{
-		Items: []runtime.Object{
-			&pods.Items[0],
-			&pods.Items[1],
-			&svc.Items[0],
+	expected := &v1.List{
+		Items: []runtime.RawExtension{
+			{Object: &pods.Items[0]},
```

Comment:
these pods are internal Pod objects... I didn't think it was expected to have internal objects in a v1.List

Reply:
Do you know what are the side effects of doing so?  The test seems to behave as expected.

Reply:
if the expectation is that `Object()` returns something you could pass to serialization without needing to do conversion, then we should be dealing in external objects

Reply:
Ah I see, if the Object is serialized directly, something will need to convert the internal objects it contains.  If the mapper is doing the conversions, perhaps it could provide an interface that takes a list of objects and puts them in the correct list - either an internal list for internal objects, and an external list for external objects?  I am guessing we want the internal/external-ness of the list to be consistent with its Items.

Reply:
Or rather the Decoder.

Reply:
> I am guessing we want the internal/external-ness of the list to be consistent with its Items.

That was my thought, but I hadn't chased it down to know. Switching the list to external while leaving the objects internal definitely seems like the type of change that could break assumptions about whether conversion was needed elsewhere.

Reply:
Sounds like this is resolved to - putting internal objects in an external list is preferred over internal objects in an internal list.

Reply:
> Sounds like this is resolved to - putting internal objects in an external list is preferred over internal objects in an internal list.

I don't see how that follows from this:

> v1.List should work with external objects (internal.List can also have external objects)

internal list can contain external objects (since serialization walks all items making sure they are converted to external if needed an no-ops on already-external ones)

v1.List should work with external objects